Sources/MockingKit/AsyncMockReference.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  AsyncTests.swift
+//
+//  Created by Tobias Boogh on 2022-05-04.
+//
+
+import Foundation
+
+/**
+ This struct is used to get a unique reference to an async function.
+ */
+public struct AsyncMockReference<Arguments, Result>: Identifiable {
```

Comment:
Do you think we need both an `AsyncMockReference` and an `AsyncThrowingMockReference`?

Reply:
I also think that I will flip the name, so that `Mock` comes first...makes it easier to find with autocomplete :)

Reply:
Nope, just remembered that the regular one is throwing as well :D